### DISCUSSION TITLE
DarkAudacity features in the code to keep

### DISCUSSION THREAD

**Discussion by LWinterberg** [audacity/audacity#2385] [Development]
* This is a spin-off discussion for #2383 

To simplify the code, I want to throw out the EXPERIMENTAL_DA flag. In the initial commit of #2383 I already removed some of the DA-project related things (logo, name changes, etc), but there's some features in it which warrant further discussion.

Some of DA's features (Mute & solo stacked vs side-by-side; Sync-Lock vs Time-lock; hiding more toolbars) already have plans for massive overhauls anyway, so I've not added them to the discussion here.

**Comment by LWinterberg** [comment]
https://github.com/audacity/audacity/blob/51677e4e32b60c27d45655a355e13ae8f470d274/libraries/lib-screen-geometry/ZoomInfo.cpp#L84-L100

I generally like the idea of disabling [snap-to](https://manual.audacityteam.org/man/selection_toolbar.html#snap) when zoomed far in, but it seems like this is a one-way operation (you zoom in, it turns off). I think it'd make more sense to have snapping disabled when zoomed in, but to turn snapping back on when zooming out.

**Reply by LWinterberg** [reply]
Actually, thinking of it some more, I think the most useful thing for snapping would be

* To bars/beats/half/quarter/8th/16th beats in a future music mode, and
* At zero crossings, except when in sample-level zoom, in a future non-music mode.

Both of which need significantly more/different code to this, so this part probably can go.

**Reply by JamesCrook** [reply]
This was based on experience.  Users got locked in 'snap to' accidentally, stuck in a mode, and didn't know what had happened.  In the hope of selecting what they wanted to select they often zoomed in to absolutely select exactly what they wanted to.  Whilst the proposed change allows them to select what they wanted to, they are still stuck in a mode, and indeed it would not be such a problem since they can achieve the edits they want.

See https://wiki.audacityteam.org/wiki/ToDo:_Unstick_%27Stuck_In_a_Mode%27 for this and other problems where users get stuck in a mode.

**Comment by LWinterberg** [comment]
https://github.com/audacity/audacity/blob/498b01c7cd1bed427814599465031cebae66ab52/src/widgets/HelpSystem.cpp#L246-L251

While Audacity currently ships an offline manual, we know from surveys that 90% of users do have internet access. I think it'd make sense to follow DA's example for the transition and switch the default to the online manual for 3.2 and see how that effects things.

**Reply by JamesCrook** [reply]
I made this change because it's less work, and I didn't want to create a special manual just for DA nor ship Audacity's manual as it were 'for DA'.  At the moment only the latest manual is online, which I think is fine, but it does mean that users who choose to stay at a particular version of Audacity will have a mismatch between their Audacity and the help they see online.  That matters less the more you push users to stay up to date.

**Comment by LWinterberg** [comment]
https://github.com/audacity/audacity/blob/498b01c7cd1bed427814599465031cebae66ab52/src/toolbars/ToolManager.cpp#L539-L544

I have no idea what the difference between the options here is, but apparently, there's a whole toolbar for transcription purposes, see also https://www.youtube.com/watch?v=9ycvGF03zQE

I don't know how much of this toolbar still remains in the code and whether it's in a working state, but I think it'd be awesome if we finally could get this in.

**Reply by Paul-Licameli** [reply]
@JamesCrook can speak for the history of decisions about Dark Audacity.

**Reply by Paul-Licameli** [reply]
The “Transcription” toolbar is the play-at-speed toolbar, with the strange name in the code for historical reasons.

The change here for Dark Audacity only changes the default arrangement of toolbars in the top dock.

**Reply by JamesCrook** [reply]
Confirming what Paul says.

The old Transcription Toolbar had some somewhat buggy additional buttons for moving between breaks between words, for example.  The play-at-speed aspect wildly useful.  The moving between breaks buttons sufficiently minority interest to warrant removing them (though the icons used, e.g. a compass, are still in the icon image).  Hence the name for the play-at-speed toolbar.
